Assets/Creatures/api/Creature.cs

```diff
@@ -32,7 +34,26 @@ protected virtual void Update()
             }
         }
 
-        protected abstract void Patrol(); // To be implemented by specific creatures
+        protected virtual void Patrol()
+        {
+            if (PatrolPoints == null || PatrolPoints.Count == 0)
+            {
+                return; // No patrol points defined
+            }
+
+            // Get the current patrol point
+            Transform patrolTarget = PatrolPoints[currentPatrolIndex];
```

Comment:
another algo I've seen: pick a patrol point in a small circle around the target, then move there, then repeat, but also define a max radius when a target spawns that they have to choose a point in. If they get to the edge, then pick a point again until they pick one that's inside their max range.

that way it's sort of random but they don't wander all over the map

Reply:
Going to kick the can on getting patrol working, should be it's own ticket (and I gots a video reference where the guy explains a cool way to do patrolling and aggro)

Reply:
https://trello.com/c/bxUoMKFq/94-add-patrolling-for-enemies